pkg/liqo-controller-manager/shadowpod-controller/shadowpod_controller.go

```diff
@@ -66,6 +60,10 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		}
 		return ctrl.Result{}, err
 	}
+	if shadowPod.Status.Phase == vkv1alpha1.ShadowPodSucceeded || shadowPod.Status.Phase == vkv1alpha1.ShadowPodFailed {
```

Comment:
Can we check the pod restart policy to take this decision?

Reply:
I agree with. For Succeeded status we don't have to check restart Policy, but for failed pod we need check restart policy. This is a good point. Thank you for caching it.

Reply:
We should restart also Succeeded pods if the restart policy is Always. I know that the recreation of the pod is very unlikely because Kubernetes should restart the pod, but it is a possibility

Reply:
I updated code to ignore the recreation only if `RestartPolicy` is set to `Never`. 
But I have a question,

- If pod has `RestartPolicy` set as `Always`, scheduled to remote cluster and it completed with `Success` status. Now `Node` is deleted this doesn't mean pod should be recreated right?
- If pod has `RestartPolicy` set as `OnFailure`, scheduled to remote cluster and it completed with `Success` status. Now `Node` is deleted this doesn't mean pod should be recreated right?
- If pod has `RestartPolicy` ser as 'Always', scheduled to remote cluster and it completed with 'Failed` stauts. Now anyway kube-scheduler will restart this pod to same node or other node. But if the node is deleted(when pod is in running state) it should be recreated as RestartPolicy is set to Always.
- if pod has `RestartPolicy` set as 'OnFailure', scheduled to remote cluster and it completed with `Failed` status. Now anyways kube-scheduler will restart this pod to same node or other node. But if the node is deleted(when pod is in running state) it should be recreated by ShadowPod Controller as RestartPolicy is set to Failed

I think if the pod status is success, irrespective of `RestartPolicy` ShadowPod Controller should not recreate the pod. But if the pod status is other than success then need to check the `RestartPolicy`, if it's set to `Never` don't recreate else recreate the pod. What do you think?

Reply:
I agree with points 2, 3, and 4, but not with the first. If I remember correctly, the only difference between `Success` and `Failed` is the exit code of the container.

If you run a pod that exits with code 0, for instance, an Ubuntu image without any command, it should alternate the `Success` and `CrashLoopBackoff` statuses since the kubelet restarts it. Following this scenario I think that we should restart the pod if it is succeeded with restart always policy set

Reply:
I’ll make the charges.

Reply:
Done. Please review. Thanks